tests/tests.go

```diff
@@ -105,15 +106,23 @@ func SetupLocalRegistry(o *option.Option) {
 		_, name, _ := strings.Cut(ref, "/")
 		// allow up to a minute for remote pulls to account for external network
 		// latency/throughput issues or throttling (default is 10 seconds)
-		// retry pull for 3 times
+		// retry pull for 3 times.
 		var session *gexec.Session
+		exitCode := -1
 		for i := 0; i < retryPull; i++ {
-			session = command.New(o, "pull", ref).WithTimeoutInSeconds(30).WithoutCheckingExitCode().Run()
-			if session.ExitCode() == 0 {
+			session = command.New(o, "pull", ref).WithoutWait().Run()
+			select {
+			case <-session.Exited:
+				exitCode = session.ExitCode()
```

Comment:
I think we still need return's in each of the cases

Reply:
https://gobyexample.com/select
from the syntax not sure we need return. can you clarify this.
Thanks

Reply:
Yeah you're right, I misread your code and thought it was still within an endless for loop for some reason. Left a couple more minor nits but this should work 👍